compiler/modules/types/common.bal

```diff
@@ -34,11 +34,18 @@ function memoSubtypeIsEmpty(Context cx, BddMemoTable memoTable, BddIsEmptyPredic
     BddMemo m;
     if mm != () {
         MemoEmpty res = mm.empty;
+        if res == "cyclic" {
+            // These are types with no non-recursive non-empty shapes (ex: A = [A];)
+            // Since we define types inductively we consider these to be empty
```

Comment:
Say we consider cyclic types to be empty because we define types inductively.

Reply:
fixed